### ISSUE TITLE
Simplify testing suite with Vitest

### ISSUE THREAD

**Issue by yeoffrey** [ianstormtaylor/superstruct#1244]
# TL;DR
The current testing suite is overly complicated. Since we've moved over to Vitest (see: #1242), we should now that package to its full potential, and simplify the testing suite all together for this project.

# More Detail
Currently the testing suite basically has a "runner" which is in `index.test.ts` which handles all the tests in the testing suite. It imports configurations from `/validation` which will then be imported with the runner (not by a typescript import, but by a module import 🤢 ) and it will execute tests.

The problem here is that is super complex. It also feels like a blackbox a bit because its not super readable and its not very strict. Now that we are using Vitest, I think theres a great opportunity to make this easier to understand and also more reliable.

# Goals
- Report code coverage
- Allow users to use the `watch` feature while testing
- Make the testing suite easy to read and add to.
